app/webpacker/components/ContactsPage/ContactForm.jsx

```diff
@@ -0,0 +1,123 @@
+import React, { useMemo, useRef, useState } from 'react';
+import {
+  Form, FormGroup, FormField, Button, Radio, Message,
+} from 'semantic-ui-react';
+import ReCAPTCHA from 'react-google-recaptcha';
+import { contactUrl } from '../../lib/requests/routes.js.erb';
+import useInputState from '../../lib/hooks/useInputState';
+import useSaveAction from '../../lib/hooks/useSaveAction';
+import I18n from '../../lib/i18n';
+import { CONTACT_TYPES, RECAPTCHA_PUBLIC_KEY } from '../../lib/wca-data.js.erb';
+import UserData from './SubForms/UserData';
+import Loading from '../Requests/Loading';
+import Wct from './SubForms/Wct';
+import Competition from './SubForms/Competition';
+
+export default function ContactForm({ userDetails }) {
+  const initialFormValues = useMemo(() => ({
+    userData: {
+      name: userDetails?.user?.name || '',
+      email: userDetails?.user?.email || '',
+    },
+  }), [userDetails]);
+  const [formValues, setFormValues] = useInputState(initialFormValues);
+  const { save, saving } = useSaveAction();
+  const [captchaError, setCaptchaError] = useState(false);
+  const recaptchaRef = useRef();
+
+  const isFormValid = (
+    formValues.contactType && formValues.userData.name && formValues.userData.email
+  );
+  const SubForm = useMemo(() => {
+    if (!formValues.contactType) return null;
+    switch (formValues.contactType) {
+      case CONTACT_TYPES[0]: // competition
```

Comment:
This should be managed better than "accidentally knowing" that index zero is `competition` (even with the code comment)

Reply:
How does the panel index do it? I vaguely remember some cool hash/object indexing there that made the code much more readable

Reply:
I've created an enum-kind of object in this class, will that be fine?

Reply:
Yup